### PR TITLE
Add google search

### PR DESCRIPTION
**Pull request by mayim-chayim**
@kavhayushar please review

### CODE REVIEW

File: extension/js/addGoogleSearch.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+function addGoogleSearch(){
+    if (window.location.href.includes("viewtopic.php")) {
+        let searchUrl = 'https://www.google.com/search?q={{value}}+site%3Aivelt.com+intitle%3A"{{title}}"'
+        let topicTitle = document.querySelectorAll('.topic-title > a')[0].text;
```
Comment (kavhayushar) [inline extension/js/addGoogleSearch.js:5]: I would move down the searchUrl declaration after the topicTtile, and change it to hardcode the title. (and remove the {{value}} placeholder.
`let searchUrl = 'https://www.google.com/search?q=site%3Aivelt.com+intitle%3A"' + topicTitle + '"'`

File: extension/js/addGoogleSearch.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        document.getElementById('search_keywords').insertAdjacentHTML('beforebegin', 
+            `
+            <input class="inputbox search tiny" style="border-width:1px;border-radius:4px" type="search" id="g_search_keywords" size="20" placeholder="גאאגל זוך אין דעם אשכול…">
+            `
+        ) 
+
+        document.getElementById('g_search_keywords').addEventListener('keydown' ,function (e) {
+            if (e.key === 'Enter') {
+                searchUrl = searchUrl.replace('{{title}}',topicTitle)
```
Comment (kavhayushar) [inline extension/js/addGoogleSearch.js:14]: Remove this line as it is already hardcoded.

File: extension/js/addGoogleSearch.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+                searchUrl = searchUrl.replace('{{value}}',document.getElementById('g_search_keywords').value)
```
Comment (kavhayushar) [inline extension/js/addGoogleSearch.js:15]: Remove this line as we are going to append it to the URL parameter below.

File: extension/js/addGoogleSearch.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                window.open(searchUrl, '_blank').focus();
```
Comment (kavhayushar) [inline extension/js/addGoogleSearch.js:16]: Add search term to the URL parameter (or if you prefer for clarity, we can save it to a variable and pass it along here.)
`window.open(searchUrl + "+" + document.getElementById("g_search_keywords").value, '_blank').focus();`

**Review by kavhayushar — CHANGES_REQUESTED** [review]
Seems to have an issue with doing a second search.
The issue is with the {{value}} replace functionality which is being replaced on the first search and no longer found on the second search.
Therefore I suggest to concat the search term when hitting the search.
See inline comments.

**Comment by kavhayushar** [comment]
All in all, nice job. I would append the changes myself but I have no access to your branch, hence no access to modify the PR.

**Review by kavhayushar — APPROVED** [review]
Works great. Thank you for your work.
Please send an invoice. 😂

**Comment by KnaperYaden** [comment]
@mayim-chayim this is only Google Search "in thread".

Can you do the same idea as a main ivelt search. Something like this:
image

(Also Google is usually spelled as גוגל, not גאאגל.)

**Comment by mayim-chayim** [comment]
> @mayim-chayim this is only Google Search "in thread".
> 
> Can you do the same idea as a main ivelt search. Something like this:

I don't think that it is so important, because you can always do a `site:ivelt.com`, to search in a thread you need an additional `intitle`, with the RTL issues adding confusion 
maybe one day I will do it
> (Also Google is usually spelled as גוגל, not גאאגל.)

Changed

**Comment by KnaperYaden** [comment]
> I don't think that it is so important, because you can always do a `site:ivelt.com`, to search in a thread you need an additional `intitle`, with the RTL issues adding confusion

Correct, but whole site search is probably needed more often (or maybe just by me...)

**Pull request merged**